extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -175,18 +192,97 @@ export class ResourceTypePickerDialog extends DialogBase {
 			const form = formBuilder.withLayout({ width: '100%' }).component();
 
 			return view.initializeModel(form).then(() => {
+				this.selectResourceType(this._resourceTypes[0]);
 				if (this._selectedResourceType) {
 					this._cardGroup.selectedCardId = this._selectedResourceType.name;
 				}
 			});
 		});
 		this._dialogObject.content = [tab];
+
+	}
+
+	private createTagsListView(): azdata.ListViewComponent {
+
+		const tags = this.getAllResourceTags();
+		if (!tags.includes('All')) {
+			tags.splice(0, 0, 'All');
+		}
+		const items: azdata.ListViewOption[] = [];
+		tags.forEach((t: string, idx: number) => {
+			items.push({
+				label: localize('deploymentDialog.tag' + idx, '{0}', t),
```

Comment:
And you also can't use concat'd strings like that for the ID - it needs to be a constant since the localization labels are bundled in as part of the build, not dynamically generated at run time.